flake.nix

```diff
@@ -4,38 +4,37 @@
   inputs.nixpkgs.url = "flake:nixpkgs";
   inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs =
-    { self
-    , nixpkgs
-    , flake-utils
-    }:
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+  }:
     flake-utils.lib.eachDefaultSystem (
-      system:
-      let
-        pkgs = import nixpkgs { inherit system; };
-        erl = pkgs.beam.packages.erlang;
-        lexical = erl.callPackage ./nix/lexical.nix {};
-      in
-      {
-        packages = {
-          inherit lexical;
+      system: let
+        pkgs = import nixpkgs {inherit system;};
+        inherit (pkgs.beam.packages) erlang_26;
 
+        mkLexical = {erlang}: erlang.callPackage ./nix/lexical.nix {};
```

Comment:
actually that isn’t correct… In the flake `erlang.callPackage` is called, so it feeds the `lexical` build derivation from `nix/lexical.nix`. however what is being overrided here is the source derivation, which doesn’t affect the resulting/child derivation…

so if you try to do this:

```nix
  {
    packages = rec {
      x = let
        y = erlang_26.callPackage ./nix/lexical.nix {};
      in y.override {erlang = pkgs.erlang_25;};
    };
  }
```

It will build successfully but starting the package with `./result/bin/start_lexical.sh iex` and issuing the OTP version:

```
iex(1)> System.otp_release
"26"
```

Doing some tests I came with this:

```
{
  packages = {
    y = erlang_25.callPackage ./nix/lexical.nix {};
    x = mkLexical {erlang = pkgs.beam.packages.erlang_25;};
    z = (erlang_26.callPackage ./nix/lexical.nix {}).overrideAttrs {erlang = pkgs.erlang_25;};
  };
}
```

And trying to get the `top_version`, only the derivation package `x` works to override the build erlang that is used 😊!

Reply:
also the package `y` works too, but it sets the erlang package mannually haha